src/conf.c

```diff
@@ -135,8 +135,10 @@ static int validate_common(cfg_t *cfg, const char *provider, int custom)
 	}
 
 	if (!custom && !cfg_getstr(cfg, "username")) {
-		cfg_error(cfg, "Missing username setting for DDNS provider %s", provider);
-		return -1;
+		if (strcmp(provider, "default@freemyip.com") != 0) {	
```

Comment:
If you really feel uncomfortable with this, then we can just tell users to configure a fake username.
So in the configuration examples, I could add a line like this:

    username = nulluser

This username won't be used by the plugin, but it will satisfy the above configuration check in conf.c, so then we can remove that if() statement.

What do you think?